scripts/autotools/CheckSSL.m4

```diff
@@ -188,18 +189,33 @@ if test "$PHP_MONGODB_SSL" = "openssl" -o "$PHP_MONGODB_SSL" = "libressl" -o "$P
     AC_SUBST(MONGOC_ENABLE_SSL_SECURE_TRANSPORT, 1)
     AC_SUBST(MONGOC_ENABLE_CRYPTO_LIBCRYPTO, 0)
     AC_SUBST(MONGOC_ENABLE_CRYPTO_COMMON_CRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_OPENSSL, 0)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 1)
+
+    PHP_MONGODB_ADD_SOURCES([src/libmongocrypt/src/], "crypto/commoncrypto.c", $PHP_MONGODB_BUNDLED_CFLAGS)
+    PHP_MONGODB_ADD_SOURCES([src/libmongocrypt/kms-message/src/], "kms_crypto_apple.c", $PHP_MONGODB_BUNDLED_CFLAGS)
   elif test "$PHP_MONGODB_SSL" = "openssl"; then
     AC_SUBST(MONGOC_ENABLE_SSL_OPENSSL, 1)
     AC_SUBST(MONGOC_ENABLE_SSL_LIBRESSL, 0)
     AC_SUBST(MONGOC_ENABLE_SSL_SECURE_TRANSPORT, 0)
     AC_SUBST(MONGOC_ENABLE_CRYPTO_LIBCRYPTO, 1)
     AC_SUBST(MONGOC_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_OPENSSL, 1)
+    AC_SUBST(MONGOCRYPT_ENABLE_CRYPTO_COMMON_CRYPTO, 0)
+
+    PHP_MONGODB_ADD_SOURCES([src/libmongocrypt/src/], "crypto/openssl.c", $PHP_MONGODB_BUNDLED_CFLAGS)
```

Comment:
@kevinAlbs: IIRC, libmongoc's own TLS-specific source files wrap themselves with compiler macros like `#ifdef MONGOC_ENABLE_SSL_OPENSSL` to ensure they're safe to compile regardless of the TLS library the project is configured to use. Is there any reason we can't do the same for libmongocrypt, given that constants like `MONGOCRYPT_ENABLE_CRYPTO_OPENSSL` exist?

@alcaeus: Were these TLS-specific source files manually excluded from the `find` commands in `config.m4` and `config.w32`? I didn't see anything in the example shell one-liner that excluded them, so I'm curious if you did so manually -- which could pose a problem if someone else were to run that command to regenerate the source file list for a particular directory.

Reply:
That seems like a good idea, created and scheduled https://jira.mongodb.org/browse/MONGOCRYPT-218

Reply:
For the `crypto/*` files, the `-maxdepth 1` argument to `find` currently excludes them. For the `kms_crypto_*` files in the other variable, we exclude the files via a `! -name 'kms_crypto_*'` argument to `find`. This makes any manual changes to the lists unnecessary, but requires maintenance if we later add any files that match the exclude pattern but should be included regardless.

I'll leave this in and have created [PHPC-1509](https://jira.mongodb.org/browse/PHPC-1509) to track this.

Reply:
SGTM. Thanks!